python/tests/test_unit_cosine_similarity.py

```diff
@@ -0,0 +1,52 @@
+import asyncio
+import numpy as np
+from pytest import raises
+from semantic_kernel.memory import VolatileMemoryStore
+from unittest import TestCase
+
+
+async def test_cosine_similarity_valid():
+    """Test the cosine similarity computation"""
+    volatile_memory_store = VolatileMemoryStore()
+    # Test case 1: Two valid embeddings
+    query_embedding = np.array([1, 0, 1])
+    collection_embeddings = np.array([[1, 0, 1], [0, 1, 0]])
+    expected_scores = np.array([1.0, 0.0])
+    scores = await volatile_memory_store.compute_similarity_scores(
+        embedding=query_embedding, embedding_array=collection_embeddings
+    )
+    np.testing.assert_allclose(scores, expected_scores)
+
+
+async def test_cosine_similarity_zero_query():
```

Comment:
To fix the warning from `pytest`, you'll need something like this (see suggestion). Don't forget to update the relevant import too.

```suggestion
@mark.asyncio
async def test_cosine_similarity_zero_query():
```

Reply:
Got it. Done